File: solutions/867. Transpose Matrix.cpp
class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& A) {
        int row = A.size(), col = A.at(0).size();
        vector<vector<int>> B(col, vector<int> (row, 0));
        for (int i = 0; i < col; i++) {
            for (int j = 0; j < row; j++) {
                B.at(i).at(j) = A.at(j).at(i);
            }
        }
        return B;
    }
};

File: solutions/867. Transpose Matrix.rs
impl Solution {
  pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
      let num_of_rows = matrix.len();
      let num_of_cols = matrix[0].len();
      let mut transposed_matrix = vec![vec![0; num_of_rows]; num_of_cols]; /* exchange row & col */
      for i in 0..num_of_rows {
          for j in 0..num_of_cols {
              transposed_matrix[j][i] = matrix[i][j];
          }
      }
      transposed_matrix
  }
}
